Framework/Algorithms/src/FindPeaks.cpp

```diff
@@ -462,8 +463,12 @@ void FindPeaks::findPeaksUsingMariscotti() {
   const int blocksize = static_cast<int>(smoothedData->blocksize());
 
   for (int k = start; k < end; ++k) {
-    const auto &S = smoothedData->y(k);
-    const auto &F = smoothedData->e(k);
+    int ws_i(k);
+    if (singleSpectrum)
+      ws_i = 0;
+
+    const auto &S = smoothedData->y(ws_i);
+    const auto &F = smoothedData->e(ws_i);
```

Comment:
I don't understand what is happening here, why was this changed?

Reply:
The old way to find peaks in automatic way includes creating a MatrixWorkspace with same number of histograms and vectors of X to calculate derivatives.  If FindPeaks only works on 1 spectrum, it wastes resources to do such.  So instead, my solution is to create a 1-spectrum MatrixWorkspace.

Reply:
Ok that makes sense :+1:. However below you duplicated a lot of code below for this purpose (calculating the derivative), I think that can and should be avoided to avoid maintenance nightmares.

If you observe an actual speedup I think that is also worth mentioning in the release notes.

Reply:
I refactored all the duplicate codes (hopefully).  On the other hand, I don't think that the speedup can be significant unless the workspace to find peaks has huge number of spectra.